src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchOptionsFactoryTests.cs

```diff
@@ -110,6 +110,41 @@ public void GivenACount_WhenCreated_ThenCorrectMaxItemCountShouldBeSet()
             Assert.Equal(5, options.MaxItemCount);
         }
 
+        [Fact]
+        public void GivenACountWithValueZero_WhenCreated_ThenCorrectMaxItemCountShouldBeSet()
+        {
+            const ResourceType resourceType = ResourceType.Encounter;
+            var queryParameters = new[]
+            {
+               Tuple.Create("_count", "0"),
+            };
+
+            SearchOptions options = CreateSearchOptions(
+            resourceType: resourceType.ToString(),
+            queryParameters: queryParameters);
+
+            Assert.NotNull(options);
+            Assert.Equal(10, options.MaxItemCount);
+        }
+
+        [Theory]
+        [InlineData("a")]
+        [InlineData("1.1")]
+        public void GivenACountWithInvalidValue_WhenCreated_ThenExceptionShouldBeThrown(string value)
+        {
+            const ResourceType resourceType = ResourceType.Encounter;
+            var queryParameters = new[]
+            {
+               Tuple.Create("_count", value),
+            };
+
+            System.FormatException exception = Assert.Throws<System.FormatException>(() => CreateSearchOptions(
+            resourceType: resourceType.ToString(),
+            queryParameters: queryParameters));
+
+            Assert.Equal("Input string was not in a correct format.", exception.Message);
```

Comment:
The expected string doesn't match the actual string.

Reply:
As discussed we have analyzed to check type of exception instead of checking actual exception message string